PROTOCOL.md

```diff
@@ -137,6 +145,10 @@ Direction: **Server -> Client**
 
 Operation execution error(s) triggered by the `Next` message happening before the actual execution, usually due to validation errors.
 
+A client should be prepared to receive (and ignore) IDs for operations for which it has previously sent a `Complete` message to the server.
+
+Note: Errors that occur during regular processing should result in a `Next` message with a non-empty `errors` member in its `payload` followed by a `Complete` message.
```

Comment:
I am working with an implementation which sends Error messages whenever an error occurs, even in a middle of a stream.  I'm trying to clarify what is the proper course of action.
Do I understand correctly that after the initial verification is complete and exectcution has started errors should always be part of the `payload`?  that the `error` message is reserved for initial verification only?

Reply:
Yes, the `Error` message is for operation validation and any other logic the implementor might add during the subscribe, the operation execution didn't occur.

The `errors` field in the `Next` message is rather an execution error; meaning, the operation execution happened and there's errors in the result. One example of one such graceful error is being subscribed to chat messages, you might have problems receiving a few messages, while the others keep coming.

Reply:
Thanks.  I'll try to make that point differently.

Reply:
> I am working with an implementation which sends Error messages whenever an error occurs, even in a middle of a stream.

I'd say it can also happen in the middle of the stream, why not, it's considered fatal. Would appreciate it if you mention this within your point as well, thanks!

Reply:
_Follow up on https://github.com/enisdenjo/graphql-ws/pull/342#discussion_r835193315_

My thinking was through the implementation of `graphql-ws`'s server, where this cannot happen. But this does not constitute that other implementors should be advised against doing so.

Reply:
Ah, okay, I already pushed a change, will edit:  So to clarify:
Sending an Error message in the middle of the stream is a permissible way to send a fatal error?  Rougly equivalent to a Next followed by Complete?

Reply:
Exactly! Sorry for the inconvenience. 😬